distributed/deploy/cluster.py

```diff
@@ -579,3 +588,59 @@ def __eq__(self, other):
 
     def __hash__(self):
         return id(self)
+
+    async def _wait_for_workers(self, n_workers=0, timeout=None):
+        await self.update_scheduler_info()
+        if timeout:
+            deadline = time() + parse_timedelta(timeout)
+        else:
+            deadline = None
+
+        def running_workers(info):
+            return len(
+                [
+                    ws
+                    for ws in info["workers"].values()
+                    if ws["status"] == Status.running.name
+                ]
+            )
+
+        while n_workers and running_workers(self.scheduler_info) < n_workers:
+            if deadline and time() > deadline:
+                raise TimeoutError(
+                    "Only %d/%d workers arrived after %s"
+                    % (running_workers(self.scheduler_info), n_workers, timeout)
+                )
+            await asyncio.sleep(0.1)
+
+            await self.update_scheduler_info()
+
+    async def update_scheduler_info(self) -> None:
+        """Send comm to scheduler requesting information and update scheduler_info accordingly"""
+        await self._scheduler_info_comm.write({"op": "identity"})
+        self.scheduler_info = SchedulerInfo(await self._scheduler_info_comm.read())
```

Comment:
You can simply do

`self.scheduler_info = SchedulerInfo(await self._scheduler.identity())`

This will allow this connection to be reused and handles the write/read part of this call. Specifically you will not need to manage the lifecycle of the comm yourself.

FWIW I think the same _should_ be done with the `_watch_worker_status_comm`. I don't see a reason why these calls would deserve a dedicated connection. I assume `_watch_worker_status_comm` was introduced *before* we had a connection pool

Reply:
Hi @fjetter, 
I get an attribute error when making this change and using a LocalCluster. 

It was quite a while ago that I wrote this code, but I'm pretty sure the reason I went down the creating-a-comm route was because clusters don't (necessarily?) have a _scheduler attribute. 

Am I missing something? Should there be a _scheduler attribute?

Reply:
Sorry, this should've been `self.scheduler_comm`.

This should always be set if a cluster is already started. The implementation is a bit messy, though. `Cluster` doesn't actually set it but `SpecCluster` does... regardless, having this set is a requirement since we're using this in many other places as well, i.e. it is safe to assume that subclasses implement this as well

Reply:
That worked! I've removed all the messy comm stuff now :)
Thank you :)